streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -309,7 +315,349 @@
         final ConsumedInternal<K, V> consumedInternal =
                 new ConsumedInternal<>(Consumed.with(materializedInternal.keySerde(), materializedInternal.valueSerde()));
 
-        return internalStreamsBuilder.table(topic, consumedInternal, materializedInternal);
+        return (KTable<K, V>) internalStreamsBuilder.table(topic, consumedInternal, materializedInternal, StateStoreType.KEY_VALUE_STORE);
+    }
+
+    /**
+     * Create a time windowed {@link KTable} for the specified topic.
+     * The {@code "auto.offset.reset"} strategy, {@link TimestampExtractor}, windowed key deserializer and value deserializers
+     * are defined by the options in {@link Consumed} are used.
+     * Input {@link KeyValue records} with {@code null} key will be dropped.
+     * <p>
+     * Note that the specified input topic must be partitioned by windowed key.
+     * If this is not the case the returned windowed {@link KTable} will be corrupted.
+     * <p>
+     * The resulting windowed {@link KTable} will be materialized in a local {@link WindowStore} using the given
+     * {@code Materialized} instance.
+     * However, no internal changelog topic is created since the original input windowed topic is supposed to be generated by
+     * another Kafka Streams application (cf. methods of {@link TimeWindowedKStream} that return a windowed {@link KTable}).
+     *
+     * </pre>
+     * To query the local {@link WindowStore} it must be obtained via
+     * {@link KafkaStreams#store(String, QueryableStoreType) KafkaStreams#store(...)}:
+     * <pre>{@code
+     * KafkaStreams streams = ...
+     * ReadOnlyWindowStore<String, Long> localStore = streams.store(queryableStoreName, QueryableStoreTypes.<String, Long>WindowStore());
+     * String key = "some-key";
+     * Long start = 1000L; // lookup start timestamp in milliseconds
+     * Long end = 2000L; // lookup end timestamp in milliseconds
+     * Long valueForKey = localStore.fetch(key, start, end); // key must be local (application state is shared over all running Kafka Streams instances)
+     * }</pre>
+     *
+     * For non-local keys, a custom RPC mechanism must be implemented using {@link KafkaStreams#allMetadata()} to
+     * query the value of the key on a parallel running instance of your Kafka Streams application.
+     *
+     * @param topic              the topic name; cannot be {@code null}
+     * @param consumed           the instance of {@link Consumed} used to define optional parameters; It cannot be {@code null}, and within it, the windowed key serde could not be {@code null}
+     * @param materialized       the instance of {@link Materialized} used to materialize a state store; It cannot be {@code null}, and within it, the store supplier and the key serde could not be {@code null}
+     * @return windowed {@link KTable} for the specified topic
+     */
+    public synchronized <K, V> KTable<Windowed<K>, V> timeWindowedTable(final String topic,
```

Comment:
the method name changed to `windowedTable` and `windowSize` parameter is missing